nilearn/decomposition/dict_learning.py

```diff
@@ -0,0 +1,202 @@
+                        t_r=t_r,
+                        keep_data_mem=True,
+                        standardize=standardize)
+        self.reduction = reduction
+        # Setting n_jobs = 1 as it is slower otherwise
+        MiniBatchDictionaryLearning.__init__(self, n_components=n_components, alpha=alpha,
+                                             n_iter=n_iter, batch_size=batch_size,
+                                             fit_algorithm='lars',
+                                             transform_algorithm='lasso_lars',
+                                             transform_alpha=alpha,
+                                             verbose=max(0, verbose - 1),
+                                             random_state=random_state,
+                                             shuffle=True,
+                                             n_jobs=1)
+
+    def _init_dict(self, imgs, y=None, confounds=None):
+        CanICA.fit(self, imgs, y=y, confounds=confounds)
+        self.data_flat_ = np.concatenate(self.data_flat_, axis=0)
+        if self.verbose:
+            print('[DictLearning] Learning time serie')
+        ridge = Ridge(alpha=1e-6, fit_intercept=None)
+        ridge.fit(self.components_.T, self.data_flat_.T)
+        self.dict_init = ridge.coef_.T
+        S = np.sqrt(np.sum(self.dict_init ** 2, axis=0))
+        self.dict_init /= S[np.newaxis, :]
+
+    def fit(self, imgs, y=None, confounds=None):
+        """Compute the mask and the ICA maps across subjects
+
+        Parameters
+        ----------
+        imgs: list of Niimg-like objects
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Data on which PCA must be calculated. If this is a list,
+            the affine is considered the same for all.
+
+        confounds: CSV file path or 2D matrix
+            This parameter is passed to nilearn.signal.clean. Please see the
+            related documentation for details
+        """
+        self._init_dict(imgs, y, confounds)
+
+        if self.verbose:
+            print('[DictLearning] Learning dictionary')
+        MiniBatchDictionaryLearning.fit(self, self.data_flat_.T)
+        if self.reduction:
+            if self.verbose:
+                print('[DictLearning] Reducing the dictionary')
+            pca = RandomizedPCA(n_components=self.n_components, iterated_power=0,
+                                random_state=self.random_state)
+            self.components_ = pca.fit_transform(self.components_)
+            data_trans = pca.transform(self.data_flat_.T)
+        else:
+            data_trans = self.data_flat_.T
+        if self.verbose:
+            print('')
+            print('[DictLearning] Learning code')
+        self.components_ = MiniBatchDictionaryLearning.transform(self, data_trans).T
+        self.components_ = as_ndarray(self.components_)
+        # flip signs in each component so that peak is +ve
```

Comment:
+ve ?
If I undestand correctly, you compare the l1 norm of positive and negative parts, which feels right.